doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1791 @@
+
+=item ECDSA_sign, ECDSA_sign_ex, ECDSA_sign_setup, ECDSA_do_sign, ECDSA_do_sign_ex,
+ECDSA_verify and ECDSA_do_verify
+
+See L</Deprecated low-level signing functions>.
+
+=item ECDSA_size
+
+Applications should use L<EVP_PKEY_size(3)>.
+
+=item EC_GF2m_simple_method, EC_GFp_mont_method, EC_GFp_nist_method,
+EC_GFp_nistp224_method, EC_GFp_nistp256_method, EC_GFp_nistp521_method and
+EC_GFp_simple_method
+
+There are no replacements for these functions. Applications should rely on the
+library automatically assigning a suitable method internally when an EC_GROUP
+is constructed.
+
+=item EC_GROUP_clear_free
+
+Use L<EC_GROUP_free(3)> instead.
+
+=item EC_GROUP_get_curve_GF2m, EC_GROUP_get_curve_GFp
+
+=item EC_GROUP_set_curve_GF2m, EC_GROUP_set_curve_GFp
+
+Applications should use L<EC_GROUP_get_curve(3)> and L<EC_GROUP_set_curve(3)>.
+
+=item EC_GROUP_have_precompute_mult, EC_GROUP_precompute_mult and EC_KEY_precompute_mult
+
+These functions are not widely used. Applications should instead switch to
+named curves which OpenSSL has hardcoded lookup tables for.
+
+=item EC_GROUP_new, EC_GROUP_method_of, EC_POINT_method_of
+
+EC_METHOD is now an internal-only concept and a suitable EC_METHOD is assigned
+internally without application intervention.
+Users of EC_GROUP_new() should switch to a different suitable constructor.
+
+=item EC_KEY_can_sign
+
+Applications should use L<EVP_PKEY_can_sign(3)> instead.
+
+=item EC_KEY_check_key
+
+See L</Deprecated low-level validation functions>
+
+=item EC_KEY_set_flags, EC_KEY_get_flags and EC_KEY_clear_flags
+
+See L<EVP_PKEY-EC(7)/Common EC parameters> which handles flags as seperate
+parameters for B<OSSL_PKEY_PARAM_EC_POINT_CONVERSION_FORMAT>,
+B<OSSL_PKEY_PARAM_EC_GROUP_CHECK_TYPE>, B<OSSL_PKEY_PARAM_EC_ENCODING>,
+B<OSSL_PKEY_PARAM_USE_COFACTOR_ECDH> and 
+B<OSSL_PKEY_PARAM_EC_INCLUDE_PUBLIC>.
+See also L<EVP_PKEY-EC(7)/EXAMPLES>
+
+=item EC_KEY_dup, EC_KEY_copy
+
+There is no direct replacement. Applications may use L<EVP_PKEY_copy_parameters(3)>
+and L<EVP_PKEY_dup(3)> instead.
+
+=item EC_KEY_decoded_from_explicit_params
+
+There is no replacement.
+
+=item EC_KEY_generate_key
+
+See L</Deprecated low-level key generation functions>.
+
+=item EC_KEY_get0_group, EC_KEY_get0_private_key, EC_KEY_get0_public_key,
+EC_KEY_get_conv_form and EC_KEY_get_enc_flags
+
+See L</Deprecated low-level key parameter getters>.
+
+=item EC_KEY_get0_engine, EC_KEY_get_default_method, EC_KEY_get_method,
+EC_KEY_new_method, EC_KEY_get_ex_data, EC_KEY_OpenSSL, EC_KEY_set_ex_data,
+EC_KEY_set_default_method, EC_KEY_METHOD_*, EC_KEY_set_method
+
+See L</Providers are a replacement for engines and low-level method overrides>
+
+=item EC_METHOD_get_field_type
+
+Use L<EC_GROUP_get_field_type(3)> instead.
+See L</Providers are a replacement for engines and low-level method overrides>
+
+=item EC_KEY_key2buf, EC_KEY_oct2key, EC_KEY_oct2priv, EC_KEY_priv2buf and EC_KEY_priv2oct
+
+There are no replacements for these.
```

Comment:
Hmm. Should there be?

Reply:
@romen ?

Reply:
Not sure what to say: the decoder/encoder code should already provide this functionality for provider-native keys,  right?

Any more direct replacement would still be deprecated as it would need to use EC_KEY type, right?

What am I missing?

Reply:
Me neither..